tools/nimbus-gradle-plugin/src/main/groovy/org/mozilla/appservices/tooling/nimbus/NimbusGradlePlugin.groovy

```diff
@@ -363,17 +363,19 @@ class NimbusPlugin implements Plugin<Project> {
             println args
         }
 
-        if(variant.hasProperty('registerJavaGeneratingTask')) {
+        if(variant.metaClass.respondsTo(variant, 'registerJavaGeneratingTask', Task, File)) {
             variant.registerJavaGeneratingTask(generateTask, new File(outputDir))
         }
 
         def generateSourcesTask = project.tasks.findByName("generate${variant.name.capitalize()}Sources")
         if (generateSourcesTask != null) {
             generateSourcesTask.dependsOn(generateTask)
         } else {
-            project.tasks.findAll().stream().filter({ task ->
-                task.name.startsWith("compile")
-            }).forEach({ task ->
+            project.tasks.findAll().stream()
+            .filter({ task ->
+                return task.name.contains("compile")
```

Comment:
Ah I understand why this branch never worked.

Reply:
Also yep 😬